{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import VCF\n",
    "import warnings\n",
    "warnings.simplefilter(action='ignore', category=FutureWarning)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: 'artificalVCF/egfr_singleHit_00.vcf'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-51-fc74d72b1eca>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0megfr_singleHit_00\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mVCF\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdataframe\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'artificalVCF/egfr_singleHit_00.vcf'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0megfr_singleHit_00\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/code/pyPackaging_scratch/VCF.py\u001b[0m in \u001b[0;36mdataframe\u001b[0;34m(filename, large)\u001b[0m\n\u001b[1;32m     72\u001b[0m         \u001b[0mcomp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'gzip'\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mfilename\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mendswith\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'.gz'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     73\u001b[0m         \u001b[0;31m# Count how many comment lines should be skipped.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 74\u001b[0;31m         \u001b[0mcomments\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_count_comments\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilename\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     75\u001b[0m         \u001b[0;31m# Return a simple DataFrame without splitting the INFO column.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     76\u001b[0m         return pd.read_table(filename, compression=comp, skiprows=comments,\n",
      "\u001b[0;32m~/code/pyPackaging_scratch/VCF.py\u001b[0m in \u001b[0;36m_count_comments\u001b[0;34m(filename)\u001b[0m\n\u001b[1;32m    154\u001b[0m     \u001b[0mcomments\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    155\u001b[0m     \u001b[0mfn_open\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgzip\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mopen\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mfilename\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mendswith\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'.gz'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 156\u001b[0;31m     \u001b[0;32mwith\u001b[0m \u001b[0mfn_open\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilename\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mfh\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    157\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mline\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mfh\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    158\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mline\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstartswith\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'#'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: 'artificalVCF/egfr_singleHit_00.vcf'"
     ]
    }
   ],
   "source": [
    "egfr_singleHit_00 = VCF.dataframe('artificalVCF/egfr_singleHit_00.vcf')\n",
    "egfr_singleHit_00"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>BRAF</th>\n",
       "      <th>EGFR</th>\n",
       "      <th>KRAS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>artificalVCF/B1_1000</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>artificalVCF/B1_1001</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>artificalVCF/B1_1002</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>artificalVCF/B1_1003</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>artificalVCF/B1_1004</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>artificalVCF/E1_1000</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>artificalVCF/E1_1001</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>artificalVCF/E1_1002</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>artificalVCF/E1_1003</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>artificalVCF/E1_1004</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>artificalVCF/K1_1000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>artificalVCF/K1_1001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>artificalVCF/K1_1002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>artificalVCF/K1_1003</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>artificalVCF/K1_1004</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Unnamed: 0  BRAF  EGFR  KRAS\n",
       "0   artificalVCF/B1_1000     1     0     0\n",
       "1   artificalVCF/B1_1001     1     0     0\n",
       "2   artificalVCF/B1_1002     1     0     0\n",
       "3   artificalVCF/B1_1003     1     0     0\n",
       "4   artificalVCF/B1_1004     1     0     0\n",
       "5   artificalVCF/E1_1000     0     1     0\n",
       "6   artificalVCF/E1_1001     0     1     0\n",
       "7   artificalVCF/E1_1002     0     1     0\n",
       "8   artificalVCF/E1_1003     0     1     0\n",
       "9   artificalVCF/E1_1004     0     1     0\n",
       "10  artificalVCF/K1_1000     0     0     1\n",
       "11  artificalVCF/K1_1001     0     0     1\n",
       "12  artificalVCF/K1_1002     0     0     1\n",
       "13  artificalVCF/K1_1003     0     0     1\n",
       "14  artificalVCF/K1_1004     0     0     1"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "t = pd.read_csv('/Users/lincoln.harris/code/cerebra/cerebra/wrkdir/geneCellMutationCounts_artifical_TEST.csv')\n",
    "t"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>11874</td>\n",
       "      <td>12227</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"DDX11L1\"; gene_name \"DDX11L1\"; transc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>12613</td>\n",
       "      <td>12721</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"DDX11L1\"; gene_name \"DDX11L1\"; transc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>13221</td>\n",
       "      <td>14409</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"DDX11L1\"; gene_name \"DDX11L1\"; transc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>14362</td>\n",
       "      <td>14829</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>14970</td>\n",
       "      <td>15038</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>15796</td>\n",
       "      <td>15947</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>16607</td>\n",
       "      <td>16765</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>16858</td>\n",
       "      <td>17055</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>17233</td>\n",
       "      <td>17368</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"MIR6859-3\"; gene_name \"MIR6859-3\"; tr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"MIR6859-2\"; gene_name \"MIR6859-2\"; tr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"MIR6859-1\"; gene_name \"MIR6859-1\"; tr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"MIR6859-4\"; gene_name \"MIR6859-4\"; tr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>17606</td>\n",
       "      <td>17742</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>17915</td>\n",
       "      <td>18061</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>18268</td>\n",
       "      <td>18366</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>24738</td>\n",
       "      <td>24891</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>29321</td>\n",
       "      <td>29370</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>30366</td>\n",
       "      <td>30503</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"MIR1302-2\"; gene_name \"MIR1302-2\"; tr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>30366</td>\n",
       "      <td>30503</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"MIR1302-11\"; gene_name \"MIR1302-11\"; ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>30366</td>\n",
       "      <td>30503</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"MIR1302-9\"; gene_name \"MIR1302-9\"; tr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>30366</td>\n",
       "      <td>30503</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"MIR1302-10\"; gene_name \"MIR1302-10\"; ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>34611</td>\n",
       "      <td>35174</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"FAM138A\"; gene_name \"FAM138A\"; transc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>34611</td>\n",
       "      <td>35174</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"FAM138C\"; gene_name \"FAM138C\"; transc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>34611</td>\n",
       "      <td>35174</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"FAM138F\"; gene_name \"FAM138F\"; transc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>35277</td>\n",
       "      <td>35481</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"FAM138A\"; gene_name \"FAM138A\"; transc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>35277</td>\n",
       "      <td>35481</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"FAM138C\"; gene_name \"FAM138C\"; transc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>35277</td>\n",
       "      <td>35481</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"FAM138F\"; gene_name \"FAM138F\"; transc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>35721</td>\n",
       "      <td>36081</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"FAM138A\"; gene_name \"FAM138A\"; transc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>unknown</td>\n",
       "      <td>exon</td>\n",
       "      <td>35721</td>\n",
       "      <td>36081</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"FAM138C\"; gene_name \"FAM138C\"; transc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066011</th>\n",
       "      <td>ERCC-00116</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>1991</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00116\"; transcript_id \"DQ668367\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066012</th>\n",
       "      <td>ERCC-00117</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>1136</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00117\"; transcript_id \"DQ459412\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066013</th>\n",
       "      <td>ERCC-00120</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>536</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00120\"; transcript_id \"DQ854992\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066014</th>\n",
       "      <td>ERCC-00123</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>1022</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00123\"; transcript_id \"DQ516782\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066015</th>\n",
       "      <td>ERCC-00126</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>1118</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00126\"; transcript_id \"DQ459427\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066016</th>\n",
       "      <td>ERCC-00130</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>1059</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00130\"; transcript_id \"EF011072\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066017</th>\n",
       "      <td>ERCC-00131</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>771</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00131\"; transcript_id \"DQ855003\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066018</th>\n",
       "      <td>ERCC-00134</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>274</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00134\"; transcript_id \"DQ516739\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066019</th>\n",
       "      <td>ERCC-00136</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>1033</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00136\"; transcript_id \"EF011063\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066020</th>\n",
       "      <td>ERCC-00137</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>537</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00137\"; transcript_id \"DQ855000\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066021</th>\n",
       "      <td>ERCC-00138</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>1024</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00138\"; transcript_id \"DQ516777\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066022</th>\n",
       "      <td>ERCC-00142</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>493</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00142\"; transcript_id \"DQ883646\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066023</th>\n",
       "      <td>ERCC-00143</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>784</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00143\"; transcript_id \"DQ668362\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066024</th>\n",
       "      <td>ERCC-00144</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>538</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00144\"; transcript_id \"DQ854995\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066025</th>\n",
       "      <td>ERCC-00145</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>1042</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00145\"; transcript_id \"DQ875386\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066026</th>\n",
       "      <td>ERCC-00147</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>1023</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00147\"; transcript_id \"DQ516790\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066027</th>\n",
       "      <td>ERCC-00148</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>494</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00148\"; transcript_id \"DQ883642\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066028</th>\n",
       "      <td>ERCC-00150</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>743</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00150\"; transcript_id \"DQ883659\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066029</th>\n",
       "      <td>ERCC-00154</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>537</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00154\"; transcript_id \"DQ854997\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066030</th>\n",
       "      <td>ERCC-00156</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>494</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00156\"; transcript_id \"DQ883643\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066031</th>\n",
       "      <td>ERCC-00157</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>1019</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00157\"; transcript_id \"DQ839618\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066032</th>\n",
       "      <td>ERCC-00158</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>1027</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00158\"; transcript_id \"DQ516795\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066033</th>\n",
       "      <td>ERCC-00160</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>743</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00160\"; transcript_id \"DQ883658\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066034</th>\n",
       "      <td>ERCC-00162</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>523</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00162\"; transcript_id \"DQ516750\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066035</th>\n",
       "      <td>ERCC-00163</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>543</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00163\"; transcript_id \"DQ668359\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066036</th>\n",
       "      <td>ERCC-00164</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>1022</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00164\"; transcript_id \"DQ516779\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066037</th>\n",
       "      <td>ERCC-00165</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>872</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00165\"; transcript_id \"DQ668363\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066038</th>\n",
       "      <td>ERCC-00168</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>1024</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00168\"; transcript_id \"DQ516776\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066039</th>\n",
       "      <td>ERCC-00170</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>1023</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00170\"; transcript_id \"DQ516773\";</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066040</th>\n",
       "      <td>ERCC-00171</td>\n",
       "      <td>ERCC</td>\n",
       "      <td>exon</td>\n",
       "      <td>1</td>\n",
       "      <td>505</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ERCC-00171\"; transcript_id \"DQ854994\";</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1066041 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  0        1     2      3      4         5  6  7  \\\n",
       "0              chr1  unknown  exon  11874  12227         .  +  .   \n",
       "1              chr1  unknown  exon  12613  12721         .  +  .   \n",
       "2              chr1  unknown  exon  13221  14409         .  +  .   \n",
       "3              chr1  unknown  exon  14362  14829         .  -  .   \n",
       "4              chr1  unknown  exon  14970  15038         .  -  .   \n",
       "5              chr1  unknown  exon  15796  15947         .  -  .   \n",
       "6              chr1  unknown  exon  16607  16765         .  -  .   \n",
       "7              chr1  unknown  exon  16858  17055         .  -  .   \n",
       "8              chr1  unknown  exon  17233  17368         .  -  .   \n",
       "9              chr1  unknown  exon  17369  17436         .  -  .   \n",
       "10             chr1  unknown  exon  17369  17436         .  -  .   \n",
       "11             chr1  unknown  exon  17369  17436         .  -  .   \n",
       "12             chr1  unknown  exon  17369  17436         .  -  .   \n",
       "13             chr1  unknown  exon  17606  17742         .  -  .   \n",
       "14             chr1  unknown  exon  17915  18061         .  -  .   \n",
       "15             chr1  unknown  exon  18268  18366         .  -  .   \n",
       "16             chr1  unknown  exon  24738  24891         .  -  .   \n",
       "17             chr1  unknown  exon  29321  29370         .  -  .   \n",
       "18             chr1  unknown  exon  30366  30503         .  +  .   \n",
       "19             chr1  unknown  exon  30366  30503         .  +  .   \n",
       "20             chr1  unknown  exon  30366  30503         .  +  .   \n",
       "21             chr1  unknown  exon  30366  30503         .  +  .   \n",
       "22             chr1  unknown  exon  34611  35174         .  -  .   \n",
       "23             chr1  unknown  exon  34611  35174         .  -  .   \n",
       "24             chr1  unknown  exon  34611  35174         .  -  .   \n",
       "25             chr1  unknown  exon  35277  35481         .  -  .   \n",
       "26             chr1  unknown  exon  35277  35481         .  -  .   \n",
       "27             chr1  unknown  exon  35277  35481         .  -  .   \n",
       "28             chr1  unknown  exon  35721  36081         .  -  .   \n",
       "29             chr1  unknown  exon  35721  36081         .  -  .   \n",
       "...             ...      ...   ...    ...    ...       ... .. ..   \n",
       "1066011  ERCC-00116     ERCC  exon      1   1991  0.000000  +  .   \n",
       "1066012  ERCC-00117     ERCC  exon      1   1136  0.000000  +  .   \n",
       "1066013  ERCC-00120     ERCC  exon      1    536  0.000000  +  .   \n",
       "1066014  ERCC-00123     ERCC  exon      1   1022  0.000000  +  .   \n",
       "1066015  ERCC-00126     ERCC  exon      1   1118  0.000000  +  .   \n",
       "1066016  ERCC-00130     ERCC  exon      1   1059  0.000000  +  .   \n",
       "1066017  ERCC-00131     ERCC  exon      1    771  0.000000  +  .   \n",
       "1066018  ERCC-00134     ERCC  exon      1    274  0.000000  +  .   \n",
       "1066019  ERCC-00136     ERCC  exon      1   1033  0.000000  +  .   \n",
       "1066020  ERCC-00137     ERCC  exon      1    537  0.000000  +  .   \n",
       "1066021  ERCC-00138     ERCC  exon      1   1024  0.000000  +  .   \n",
       "1066022  ERCC-00142     ERCC  exon      1    493  0.000000  +  .   \n",
       "1066023  ERCC-00143     ERCC  exon      1    784  0.000000  +  .   \n",
       "1066024  ERCC-00144     ERCC  exon      1    538  0.000000  +  .   \n",
       "1066025  ERCC-00145     ERCC  exon      1   1042  0.000000  +  .   \n",
       "1066026  ERCC-00147     ERCC  exon      1   1023  0.000000  +  .   \n",
       "1066027  ERCC-00148     ERCC  exon      1    494  0.000000  +  .   \n",
       "1066028  ERCC-00150     ERCC  exon      1    743  0.000000  +  .   \n",
       "1066029  ERCC-00154     ERCC  exon      1    537  0.000000  +  .   \n",
       "1066030  ERCC-00156     ERCC  exon      1    494  0.000000  +  .   \n",
       "1066031  ERCC-00157     ERCC  exon      1   1019  0.000000  +  .   \n",
       "1066032  ERCC-00158     ERCC  exon      1   1027  0.000000  +  .   \n",
       "1066033  ERCC-00160     ERCC  exon      1    743  0.000000  +  .   \n",
       "1066034  ERCC-00162     ERCC  exon      1    523  0.000000  +  .   \n",
       "1066035  ERCC-00163     ERCC  exon      1    543  0.000000  +  .   \n",
       "1066036  ERCC-00164     ERCC  exon      1   1022  0.000000  +  .   \n",
       "1066037  ERCC-00165     ERCC  exon      1    872  0.000000  +  .   \n",
       "1066038  ERCC-00168     ERCC  exon      1   1024  0.000000  +  .   \n",
       "1066039  ERCC-00170     ERCC  exon      1   1023  0.000000  +  .   \n",
       "1066040  ERCC-00171     ERCC  exon      1    505  0.000000  +  .   \n",
       "\n",
       "                                                         8  \n",
       "0        gene_id \"DDX11L1\"; gene_name \"DDX11L1\"; transc...  \n",
       "1        gene_id \"DDX11L1\"; gene_name \"DDX11L1\"; transc...  \n",
       "2        gene_id \"DDX11L1\"; gene_name \"DDX11L1\"; transc...  \n",
       "3        gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...  \n",
       "4        gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...  \n",
       "5        gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...  \n",
       "6        gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...  \n",
       "7        gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...  \n",
       "8        gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...  \n",
       "9        gene_id \"MIR6859-3\"; gene_name \"MIR6859-3\"; tr...  \n",
       "10       gene_id \"MIR6859-2\"; gene_name \"MIR6859-2\"; tr...  \n",
       "11       gene_id \"MIR6859-1\"; gene_name \"MIR6859-1\"; tr...  \n",
       "12       gene_id \"MIR6859-4\"; gene_name \"MIR6859-4\"; tr...  \n",
       "13       gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...  \n",
       "14       gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...  \n",
       "15       gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...  \n",
       "16       gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...  \n",
       "17       gene_id \"WASH7P\"; gene_name \"WASH7P\"; transcri...  \n",
       "18       gene_id \"MIR1302-2\"; gene_name \"MIR1302-2\"; tr...  \n",
       "19       gene_id \"MIR1302-11\"; gene_name \"MIR1302-11\"; ...  \n",
       "20       gene_id \"MIR1302-9\"; gene_name \"MIR1302-9\"; tr...  \n",
       "21       gene_id \"MIR1302-10\"; gene_name \"MIR1302-10\"; ...  \n",
       "22       gene_id \"FAM138A\"; gene_name \"FAM138A\"; transc...  \n",
       "23       gene_id \"FAM138C\"; gene_name \"FAM138C\"; transc...  \n",
       "24       gene_id \"FAM138F\"; gene_name \"FAM138F\"; transc...  \n",
       "25       gene_id \"FAM138A\"; gene_name \"FAM138A\"; transc...  \n",
       "26       gene_id \"FAM138C\"; gene_name \"FAM138C\"; transc...  \n",
       "27       gene_id \"FAM138F\"; gene_name \"FAM138F\"; transc...  \n",
       "28       gene_id \"FAM138A\"; gene_name \"FAM138A\"; transc...  \n",
       "29       gene_id \"FAM138C\"; gene_name \"FAM138C\"; transc...  \n",
       "...                                                    ...  \n",
       "1066011    gene_id \"ERCC-00116\"; transcript_id \"DQ668367\";  \n",
       "1066012    gene_id \"ERCC-00117\"; transcript_id \"DQ459412\";  \n",
       "1066013    gene_id \"ERCC-00120\"; transcript_id \"DQ854992\";  \n",
       "1066014    gene_id \"ERCC-00123\"; transcript_id \"DQ516782\";  \n",
       "1066015    gene_id \"ERCC-00126\"; transcript_id \"DQ459427\";  \n",
       "1066016    gene_id \"ERCC-00130\"; transcript_id \"EF011072\";  \n",
       "1066017    gene_id \"ERCC-00131\"; transcript_id \"DQ855003\";  \n",
       "1066018    gene_id \"ERCC-00134\"; transcript_id \"DQ516739\";  \n",
       "1066019    gene_id \"ERCC-00136\"; transcript_id \"EF011063\";  \n",
       "1066020    gene_id \"ERCC-00137\"; transcript_id \"DQ855000\";  \n",
       "1066021    gene_id \"ERCC-00138\"; transcript_id \"DQ516777\";  \n",
       "1066022    gene_id \"ERCC-00142\"; transcript_id \"DQ883646\";  \n",
       "1066023    gene_id \"ERCC-00143\"; transcript_id \"DQ668362\";  \n",
       "1066024    gene_id \"ERCC-00144\"; transcript_id \"DQ854995\";  \n",
       "1066025    gene_id \"ERCC-00145\"; transcript_id \"DQ875386\";  \n",
       "1066026    gene_id \"ERCC-00147\"; transcript_id \"DQ516790\";  \n",
       "1066027    gene_id \"ERCC-00148\"; transcript_id \"DQ883642\";  \n",
       "1066028    gene_id \"ERCC-00150\"; transcript_id \"DQ883659\";  \n",
       "1066029    gene_id \"ERCC-00154\"; transcript_id \"DQ854997\";  \n",
       "1066030    gene_id \"ERCC-00156\"; transcript_id \"DQ883643\";  \n",
       "1066031    gene_id \"ERCC-00157\"; transcript_id \"DQ839618\";  \n",
       "1066032    gene_id \"ERCC-00158\"; transcript_id \"DQ516795\";  \n",
       "1066033    gene_id \"ERCC-00160\"; transcript_id \"DQ883658\";  \n",
       "1066034    gene_id \"ERCC-00162\"; transcript_id \"DQ516750\";  \n",
       "1066035    gene_id \"ERCC-00163\"; transcript_id \"DQ668359\";  \n",
       "1066036    gene_id \"ERCC-00164\"; transcript_id \"DQ516779\";  \n",
       "1066037    gene_id \"ERCC-00165\"; transcript_id \"DQ668363\";  \n",
       "1066038    gene_id \"ERCC-00168\"; transcript_id \"DQ516776\";  \n",
       "1066039    gene_id \"ERCC-00170\"; transcript_id \"DQ516773\";  \n",
       "1066040    gene_id \"ERCC-00171\"; transcript_id \"DQ854994\";  \n",
       "\n",
       "[1066041 rows x 9 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hg38_gtf = pd.read_csv('/Users/lincoln.harris/code/cerebra/cerebra/wrkdir/hg38-plus.gtf', delimiter = '\\t', header = None)\n",
    "hg38_gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'DDX11L1'"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "returnStr = str(hg38_gtf.iloc[0][8])     # keep just the gene name / metadata col\n",
    "returnStr = returnStr.split(';')[1]\n",
    "returnStr = returnStr.strip(' gene_name')\n",
    "returnStr = returnStr.strip(' ')\n",
    "returnStr = returnStr.strip('\"')\n",
    "returnStr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def get_gene_name(posString):\n",
    "    \"\"\" return the gene name from a given genome position string\n",
    "       (ie. '1:21890111-21890111'), by querying the hg38-plus.gtf \"\"\"\n",
    "\n",
    "    chrom = posString.split(':')[0] # work on posString\n",
    "    posString_remove = posString.split(':')[1]\n",
    "    lPosition = posString_remove.split('-')[0] \n",
    "    rPosition = posString_remove.split('-')[1] \n",
    "\n",
    "    # work on hg38_gtf\n",
    "    chromStr = 'chr' + str(chrom)\n",
    "    hg38_gtf_filt = hg38_gtf.where(hg38_gtf[0] == chromStr).dropna()\n",
    "    hg38_gtf_filt = hg38_gtf_filt.where(hg38_gtf_filt[3] <= int(lPosition)).dropna() # lPos good\n",
    "    hg38_gtf_filt = hg38_gtf_filt.where(hg38_gtf_filt[4] >= int(rPosition)).dropna() # rPos good\n",
    "\n",
    "    try:\n",
    "        returnStr = str(hg38_gtf_filt.iloc[0][8])\t# keep just the gene name / metadata col\n",
    "        returnStr = returnStr.split(';')[1]\n",
    "        returnStr = returnStr.strip(' gene_name')\n",
    "        returnStr = returnStr.strip(' ')\n",
    "        returnStr = returnStr.strip('\"')\n",
    "    except IndexError:\n",
    "        returnStr = ''\n",
    "\n",
    "    return returnStr\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'HSPG2'"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# test case \n",
    "get_gene_name('1:21890111-21890111')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'EGFR-AS1'"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# not quite \n",
    "get_gene_name('7:55182350-55182350')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "''"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# should be EGFR\n",
    "get_gene_name('7:55152340-55152340')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "''"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# should be BRAF\n",
    "get_gene_name('7:140900700-140900700')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'EGFR'"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# why does this one work? \n",
    "get_gene_name('7:55191821-55191821')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'EGFR-AS1'"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_gene_name('7:55181377-55181379')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'BRAF'"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_gene_name('7:140753335-140753337')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'BRAF'"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_gene_name('7:140753336-140753336')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'KRAS'"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_gene_name('12:25245351-25245351')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
